### PR TITLE
Remove HOSTNAME and IP/IP6 env vars

### PR DESCRIPTION
**Pull request by bcreane**
## Description
No longer set HOSTNAME or IP/IP6 environment variables because they've been replaced by NODENAME and Node resource, respectively.
- this is an enhancement: https://github.com/projectcalico/calico-bgp-daemon/issues/18
- part of a bigger feature, e.g. https://github.com/projectcalico/calicoctl/pull/1453
- successfully ran calico_node STs for BGP
- manually verified the k8s on vagrant/coreos setup has reachability between containers on different nodes
- updated the run command's documentation which mentions these variables

I don't see a release note mentioning this change. @caseydavenport , can you confirm?

## Todos
- [x] Tests
- [x] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Putting this in next-milestone for now, I don't think we need this for v2.4.

File: master/reference/calicoctl/commands/node/run.md [hunk 1]
```diff
@@ -135,7 +135,7 @@ Enabling IPv6 forwarding
 Increasing conntrack limit
 Running the following command:
 
-docker run --net=host --privileged --name=calico-node -d --restart=always -e ETCD_SCHEME=http -e HOSTNAME=calico -e CALICO_LIBNETWORK_ENABLED=true -e ETCD_AUTHORITY=127.0.0.1:2379 -e AS= -e NO_DEFAULT_POOLS= -e ETCD_ENDPOINTS= -e IP= -e IP6= -e CALICO_NETWORKING_BACKEND=bird -v /var/run/docker.sock:/var/run/docker.sock -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /var/log/calico:/var/log/calico -v /run/docker/plugins:/run/docker/plugins quay.io/calico/node:{{site.data.versions[page.version].first.title}}
+docker run --net=host --privileged --name=calico-node -d --restart=always -e ETCD_SCHEME=http -e CALICO_LIBNETWORK_ENABLED=true -e ETCD_AUTHORITY=127.0.0.1:2379 -e AS= -e NO_DEFAULT_POOLS= -e ETCD_ENDPOINTS= -e CALICO_NETWORKING_BACKEND=bird -v /var/run/docker.sock:/var/run/docker.sock -v /var/run/calico:/var/run/calico -v /lib/modules:/lib/modules -v /var/log/calico:/var/log/calico -v /run/docker/plugins:/run/docker/plugins quay.io/calico/node:{{site.data.versions[page.version].first.title}}
```
Comment (robbrockbank) [inline master/reference/calicoctl/commands/node/run.md]: This change is unrelated to the issue no?

**Comment by robbrockbank** [comment]
I don't think the issue requires the node run documentation to be updated does it, or is that addressing a different issue?

File: master/reference/calicoctl/commands/node/run.md (hunk 1 above)
Comment (bcreane) [inline master/reference/calicoctl/commands/node/run.md]: yeah, not directly related to this PR. will remove and respin the PR.

**Comment by robbrockbank** [comment]
LGTM
